### DISCUSSION TITLE
When will vm be stopped?

### DISCUSSION THREAD

**Discussion by mafreud** [cirruslabs/tart#471] [Q&A]
Hi,
Could you please tell me under what conditions does the vm stop?
When I was using Tart VM as a self hosted runner in GH, after a few hours, it stopped.

**Comment by edigaryev** [comment]
VMs normally stop when the guest shuts down, when you send a signal to the `tart run` process,  invoke `tart stop` or simply close the GUI window.

Regarding the other reasons, consult the standard output and standard error of the `tart run` to figure out what's happening.

**Comment by fkorotkov** [comment]
There might be a lot of reasons for a VM to appear in a stopped state especially if you run a long running VM as your runner and don't use ephemeral VMs:

- VM crashed
- Inside VM it was instructed to shutdown macOS with something like `sudo shutdown -h now`
- Close window of the VM
- Restart host. Do you have auto-updates enabled on the host?

I'd suggest to try instrument your setup with some logging to have more data when it happens again. Or you can try using [our managed GitHub Actions Runners](https://tart.run/integrations/github-actions/) that are executed in ephemeral Tart VMs and where we have bunch of instrumentation and handling a lot of corner case. 😉
